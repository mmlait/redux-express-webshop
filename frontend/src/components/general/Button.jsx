import React from 'react';
import styled from 'styled-components';
import colors from '../../colors';

const ButtonWrapper = styled.button`
  background-color: ${colors.primary};
  color: ${colors.secondary};
  padding: 15px;
  min-width: 100px;
  border: none;
  border-radius: 10px;
  cursor: pointer;
  &:hover {
    background-color: ${colors.btnHover};
  }
  &:focus {
    outline: none;
  }
`

const Button = ({onclick, type, content}) => {
  return (
    <ButtonWrapper onClick={onclick} type={type}>
      {content}
    </ButtonWrapper>
  )
}

export default Button;
