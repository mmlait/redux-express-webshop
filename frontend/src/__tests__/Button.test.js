import React from 'react';
import Button from '../components/general/Button';
import { create } from 'react-test-renderer'

describe('Button', () => {
    it('should render Button component', () => {
    let tree = create(<Button />)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})
