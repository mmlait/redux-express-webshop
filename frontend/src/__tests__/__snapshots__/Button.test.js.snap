// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button component 1`] = `
<button
  className="sc-bdVaJa kXhGps"
/>
`;
